src/goto-symex/symex_assign.cpp

```diff
@@ -378,7 +472,41 @@ void symex_assignt::assign_non_struct_symbol(
     ns);
 
   if(symex_config.simplify_opt)
+  {
+    if(rhs.id() == ID_function_application)
+    {
+      INVARIANT(guard.empty(), "assignment does not have a guard");
+      INVARIANT(
+        rhs.id() == assignment.rhs.id(),
+        "rhs has same kind after renaming and field sensitivity processing");
```

Comment:
I'm not sure of the reason for these invariant, maybe they are just needed for `constant_propagate_empty_string`, in which case they should come later.